### PR TITLE
DM-12766: Add __reduce__ to CoaddDriver so to know how to pickle

### PR DESCRIPTION
**Pull request by hsinfang**
Running coaddDriver with cores >1 has been producing `MPI_Abort(MPI_COMM_WORLD, 1)`

### CODE REVIEW

File: python/lsst/pipe/drivers/coaddDriver.py [hunk 1]
```diff
@@ -89,6 +93,11 @@ def __init__(self, reuse=tuple(), **kwargs):
         self.makeSubtask("assembleCoadd")
         self.makeSubtask("detectCoaddSources")
 
+    def __reduce__(self):
+        """Pickler"""
+        return unpickle, (self.__class__, [], dict(config=self.config, name=self._name,
+                                              parentTask=self._parentTask, log=self.log))
```
Comment (PaulPrice) [inline python/lsst/pipe/drivers/coaddDriver.py]: Indent to match paren above.

File: python/lsst/pipe/drivers/coaddDriver.py [hunk 2]
```diff
@@ -75,6 +75,10 @@ def getTargetList(parsedCmd, **kwargs):
         return [(parsedCmd.id.refList, kwargs), ]
 
 
+def unpickle(factory, args, kwargs):
+    """Unpickle something by calling a factory"""
+    return factory(*args, **kwargs)
+
```
Comment (PaulPrice) [inline python/lsst/pipe/drivers/coaddDriver.py:81]: Two lines between function and class.

File: python/lsst/pipe/drivers/coaddDriver.py (hunk 1 above)
Comment (PaulPrice) [inline python/lsst/pipe/drivers/coaddDriver.py]: Where's `reuse`?

**Review by PaulPrice — CHANGES_REQUESTED** [review]
A couple of style changes requested, and one important omission: the `__reduce__` method doesn't preserve the `reuse` argument.

File: python/lsst/pipe/drivers/coaddDriver.py (hunk 1 above)
Comment (hsinfang) [reply python/lsst/pipe/drivers/coaddDriver.py]: Thanks for the catch. Adding it to MultiBandDriver too.

**Pull request merged**